### PR TITLE
PCS-87130 Removing repo json from  reposyncstate

### PR DESCRIPTION
**Pull request by mboudreau**
**What's in this PR?**
The removal of all RepoJson objects in the codebase

**Why**
RepoJson was the old way of storing RepoSyncState in a `repoSyncState` JSON column under the subscription.  This was incredibly slow, annoying to update and caused issues for larger customers.  We had to do the change over from the column to it's own table without downtime, so we slowly moved the data over to the `RepoSyncState` table while still using the same data structure as the old `repoSyncState` column, until we didn't need the column anymore and 100% relied on the new table.  However, we never removed the old data structure which made things much more complicated than needed to do and after an issue was discovered from a larger customer while updating `RepoSyncState`, we now need to remove it completely to finally end the migration process and rely 100% on proper database architecture.

**Added feature flags**
None.  Impossible to do without duplicating all code using RepoSyncState

**Affected issues**  
PCS-87130

**How has this been tested?**  
Unit tests, pollinator and manual

**Whats Next?**
Deploy then backfill customer which this has been causing issues for.

### CODE REVIEW

File: src/config/statsd.ts
Comment (joshkay10) [inline src/config/statsd.ts:24]: Good find!

File: db/config.json
Comment (joshkay10) [inline db/config.json:7]: For my benefit lolol, why 60000 and 15?

**Review by joshkay10 — APPROVED** [review]
Juicy change! Find this new way simpler to understand too so thats nice.
A second review might not be the worst thing if you can get one lololololol

File: src/backfill/prioritizer.ts
Comment (mboudreau) [reply src/backfill/prioritizer.ts:60]: I agree, it's taken from below and is a horrible standard, but as far as I can tell, this file isn't being used but I had to change it because the old pullCursor was for numbers and not strings.

I think the idea is to remove this file eventually or just fix our  backfilling mechanism

File: db/config.json
Comment (mboudreau) [reply db/config.json:7]: 15 is what we use in production, but also noticed that for some operations on my local, stuff was being bottlenecked and taking forever to backfill stuff one a single connection.  I added the acquire as I sometimes get a weird timeout bug when starting out, not sure why, but this seems to fix it after reading some stuff online.  I think we need to update to latest sequelize to fix some of these issues.

**Review by joshkay10 — APPROVED** [review]

File: src/backfill/looper/looper.ts
Comment (thombergs) [inline src/backfill/looper/looper.ts:1]: Sad to see this code go as it took care of quite some issues with ratelimiting, retrying, etc. But I never pushed for adopting this backfilling style. cc @joshkay10 is there anything we still need to fix in the backfilling mechanism to make it more stable? We can still use this for inspiration

File: src/backfill/prioritizer.ts
Comment (thombergs) [inline src/backfill/prioritizer.ts:1]: If `looper.ts` is gone, `prioritizer.ts` can go, too. Unless we want to finally refactor the backfilling mechanism

File: src/models/subscription.test.ts
Comment (thombergs) [inline src/models/subscription.test.ts:53]: Do we have a task somewhere that keeps track of adding these tests?

File: src/models/subscription.ts
Comment (thombergs) [inline src/models/subscription.ts:187]: Maybe explain why it's not necessary? Whoever looks at this TODO will ask exactly that.

**Review by thombergs — APPROVED** [review]
some small comments.

Before merging, I would like to have it run in staging / dev once and check that the Pollinator tests are still running through and manually check that the backfilling is still working, because a lot of the code touched the backfilling.

File: src/models/subscription.test.ts
Comment (mboudreau) [reply src/models/subscription.test.ts:53]: would be nice if we had a plugin that could take TODOs and add them to jira ;)

File: src/backfill/prioritizer.ts
Comment (mboudreau) [reply src/backfill/prioritizer.ts:1]: we're not using prioritizer at all it seems, I'm glad to remove it since we need to redo the backfill properly.

File: src/models/subscription.test.ts
Comment (mboudreau) [reply src/models/subscription.test.ts:53]: but yeah, I was going to add some tasks to jira

File: src/backfill/looper/looper.ts
Comment (joshkay10) [reply src/backfill/looper/looper.ts:1]: We don't have the preemptive ratelimiting smarts yet.... I still want to spike this solution and bring over a couple of the recent enhancements, it does feel like a shame to drop it without an honest go. The best backfill solution will be an ever going debate, but it's worth evaluating the effort vs reward.

**Comment by mboudreau** [comment]
@thombergs I've already deployed it to staging previously, pollinator passed and I've done manual testing of all aspects with success.  Redeploying to staging now for others to test :)

**Review by joshkay10 — DISMISSED** [review]

File: src/models/subscription.test.ts
Comment (thombergs) [inline src/models/subscription.test.ts:11]: Can the commented code here be removed completely?

File: src/sync/discovery.ts
Comment (thombergs) [inline src/sync/discovery.ts:19]: Can this code be deleted?

**Review by thombergs — APPROVED** [review]

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**
